default-fields/0000-default-fields.md

```diff
@@ -6,31 +6,206 @@
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+}
+```
+
+Field defaults are sugar for the 'real' initialiser, where values for missing fields are added with the supplied default expression.
+
+Supplied field values take precedence over field defaults:
+
+```rust
+// `b` is `false`, even though the field default is `true`
+let foo = Foo {
+    a: "Hello",
+    b: false,
+    c: 42,
+};
+```
+
+Supplied field values in functional updates take precedence over field defaults:
+
+```rust
+// `b` is `false`, even though the field default is `true`
+let foo = Foo {
+    a: "Hello",
+    c: 0,
+    ..Foo {
+        b: false
+    }
+};
+```
+
+The field default should only be used when the caller doesn't supply a value,
+to avoid unnecessarily assigning values.
+
+When deriving `Default`, field defaults are used instead of the type default.
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+}
+
+// `b` is `true`, even though `bool::default()` is `false`
+let foo = Foo::default();
+```
+
+## Allowable Values
+
+Field defaults use the same syntax as `const`s.
+So the type must be supplied, and the value must be a compile-time expression, or a call to `Default::default()`.
+Default field values are expected to be cheap to produce and have basic support for collection types like `Vec`.
 
 # Drawbacks
 [drawbacks]: #drawbacks
 
-Why should we *not* do this?
+Field defaults are limited to `const` expressions and calls to `default()`.
+This means there are values that can't be used as defaults, such as a `Vec` with 3 elements.
+
+Allowing functionality to be injected into data initialisation through abuse of `Default` means struct literal initialisation 
+is no longer guaranteed to be pure.
 
 # Alternatives
 [alternatives]: #alternatives
 
-What other designs have been considered? What is the impact of not doing this?
+## Allow arbitrary expressions instead of constants
 
-# Unresolved questions
-[unresolved]: #unresolved-questions
+Allowing arbitrary expressions as defaults would make this feature more powerful, 
+but at the expense of allowing functionality to leak into the struct's data.
+
+Limiting valid field defaults to `const`s and defaults keeps confidence that the cost of initialising a struct
+will be low.
+The same isn't true when arbitrary expressions that could reasonably panic or block on io are allowed.
+
+It could be argued that supporting `Default` is an artificial constraint that doesn't prevent arbitrary expressions.
+The difference is that `Default` has an expectation of being cheap, so using it to inject logic into field
+initialisation is an obvious code smell.
+
+Ultimately, the combination of `const`s and `Default` strikes the best balance between expressiveness of allowable
+field default values and constraints on their cost.
+When using `const` values, there is no runtime overhead for default fields.
+When using `Default`, the expected overhead is small.
```

Comment:
Sans-default is now what the RFC is propising, and `default` is in alternatives.